tools/yul_phaser/Chromosome.cpp

```diff
@@ -0,0 +1,190 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <tools/yul_phaser/Chromosome.h>
+
+#include <tools/yul_phaser/Random.h>
+
+#include <libyul/optimiser/BlockFlattener.h>
+#include <libyul/optimiser/CommonSubexpressionEliminator.h>
+#include <libyul/optimiser/ConditionalSimplifier.h>
+#include <libyul/optimiser/ConditionalUnsimplifier.h>
+#include <libyul/optimiser/ControlFlowSimplifier.h>
+#include <libyul/optimiser/DeadCodeEliminator.h>
+#include <libyul/optimiser/EquivalentFunctionCombiner.h>
+#include <libyul/optimiser/ExpressionInliner.h>
+#include <libyul/optimiser/ExpressionJoiner.h>
+#include <libyul/optimiser/ExpressionSimplifier.h>
+#include <libyul/optimiser/ExpressionSplitter.h>
+#include <libyul/optimiser/ForLoopConditionIntoBody.h>
+#include <libyul/optimiser/ForLoopConditionIntoBody.h>
+#include <libyul/optimiser/ForLoopConditionOutOfBody.h>
+#include <libyul/optimiser/ForLoopInitRewriter.h>
+#include <libyul/optimiser/FullInliner.h>
+#include <libyul/optimiser/FunctionGrouper.h>
+#include <libyul/optimiser/FunctionHoister.h>
+#include <libyul/optimiser/LoadResolver.h>
+#include <libyul/optimiser/LoopInvariantCodeMotion.h>
+#include <libyul/optimiser/OptimiserStep.h>
+#include <libyul/optimiser/RedundantAssignEliminator.h>
+#include <libyul/optimiser/Rematerialiser.h>
+#include <libyul/optimiser/SSAReverser.h>
+#include <libyul/optimiser/SSATransform.h>
+#include <libyul/optimiser/StructuralSimplifier.h>
+#include <libyul/optimiser/Suite.h>
+#include <libyul/optimiser/UnusedPruner.h>
+#include <libyul/optimiser/VarDeclInitializer.h>
+#include <libyul/optimiser/VarNameCleaner.h>
+
+#include <cassert>
+#include <sstream>
+
+using namespace std;
+using namespace solidity::yul;
+
+namespace solidity::phaser
+{
+
+Chromosome Chromosome::make_random(int _length)
+{
+	assert(_length >= 0);
+
+	vector<string> steps;
+	steps.reserve(_length);
```

Comment:
Does this save so much?

Reply:
Probably nothing measurable. 4 or 5 reallocations since there are 28 possible steps right now. It's just that not pre-allocating the memory when I know the exact size ahread of time seems kinda wrong to me :)

Should I drop it?

Reply:
Once we determine that this function is crucial for performance, we can reevaluate, but at this point, this is more noise than helpful. It might even be that the compiler/optiimzer can determine the length from looking at the loop...

Reply:
Removed this and a few other similar calls to `reserve()`. From now on I'll use it only if it's likely to affect performance.